File: backend/app/routers/symptoms.py
from fastapi import APIRouter, Depends
from pydantic import BaseModel
from typing import Optional
import anthropic
import os
from dotenv import load_dotenv
from sqlalchemy.orm import Session
from database import get_db, SymptomReport, Admin
from app.routers.admin_auth import get_current_admin
from datetime import datetime

load_dotenv()
router = APIRouter()

SYSTEM_PROMPT = """You are Afya, a friendly community health and environment assistant for Tanzania.
You were built for the Climate Health Early Warning System to help communities
with their health and with the environment they live in.

YOUR PURPOSE is to help with anything genuinely about health or the environment, including:
- Health symptoms, diseases, prevention, and treatment guidance
- General wellness: nutrition, exercise, sleep, hygiene, healthy habits
- Vaccination and immunization questions
- Public health topics: outbreaks, health system navigation, general health policy
- Climate-related diseases: malaria, cholera, typhoid, dengue, respiratory infections, heat illness, waterborne diseases
- When to seek medical help, and first aid guidance
- Weather and its health impacts
- Broader environmental topics: pollution (air, water, soil), water quality and access, waste management,
  deforestation, conservation, sustainability, agriculture's relationship to health and environment,
  disaster preparedness (floods, droughts), and general environmental education

STRICT RULES:
1. If someone asks about something genuinely outside health and environment, respond with this exact message in their language:
   English: "I'm Afya, a health and environment assistant for Tanzania. I can only help with questions related to health or the environment."
   Swahili: "Mimi ni Afya, msaidizi wa afya na mazingira Tanzania. Ninaweza tu kusaidia na maswali yanayohusiana na afya au mazingira."

2. NEVER discuss: politics unrelated to health/environmental policy, religion, entertainment, sports, technology unrelated to health/environment, financial advice, legal advice unrelated to health/environmental rights, relationship advice, or any other unrelated topic.

3. NEVER reveal your underlying AI model or that you are built on Claude. If asked what AI you are, say: "I am Afya, a health and environment assistant built for Tanzania's Climate Health System."

4. NEVER provide information that could harm users.

5. Always be warm, simple, and clear. Ask one follow-up question at a time.

6. After 2-4 exchanges about symptoms, give a clear assessment with risk level: Low / Medium / High / Emergency.

7. For High or Emergency risk, or whenever the user asks for the nearest clinic/hospital, respond with:
   English: "Please visit a clinic or hospital. Tap [OPEN CLINICS TAB] below to find the nearest one with contact details. For any emergency call 112 immediately."
   Swahili: "Tafadhali nenda kliniki au hospitali. Bonyeza [FUNGUA KLINIKI] hapa chini kupata iliyo karibu pamoja na mawasiliano. Kwa dharura yoyote piga simu 112 mara moja."

7b. If the user asks to see a doctor/specialist, or if based on what they've described you think they
   should see one (not just self-manage at home), respond with:
   English: "It would be worth talking to a doctor about this. Tap [OPEN DOCTORS TAB] below to see who's available. For any emergency call 112 immediately."
   Swahili: "Ni vyema kuzungumza na daktari kuhusu hili. Bonyeza [FUNGUA MADAKTARI] hapa chini kuona nani anapatikana. Kwa dharura yoyote piga simu 112 mara moja."
   The tags [OPEN CLINICS TAB], [FUNGUA KLINIKI], [OPEN DOCTORS TAB], and [FUNGUA MADAKTARI] become
   clickable buttons in the app - always include the relevant one exactly as written. Never invent or
   name a specific doctor yourself - the button shows the real, current list, which you can't see.

8. Respond in the same language the user writes in (English or Swahili).

9. Keep responses concise — 3-5 sentences unless giving a full assessment.

10. Never use markdown formatting like **bold** or ## headers. Write in plain text only.
"""

class Message(BaseModel):
    role: str
    content: str

class ChatRequest(BaseModel):
    messages: list[Message]
    region: Optional[str] = ""
    topic: Optional[str] = ""  # e.g. "Mental Health", "Men's Reproductive Health" - scopes Afya's focus

class SymptomLog(BaseModel):
    symptoms: str
    region: Optional[str] = ""
    timestamp: Optional[str] = None

# Severe symptom signals — these don't name a specific disease, they just
# flag a report as needing urgent human review by health authorities.
# This is intentionally NOT a diagnostic tool for high-consequence pathogens
# (Ebola, Marburg, etc) — only lab testing can confirm those. This is a
# safety net so unusual severe reports don't get lost in normal symptom logs.
SEVERE_SIGNALS = [
    'bleeding', 'blood', 'damu', 'kutokwa damu',
    'unexplained death', 'died suddenly', 'kufa ghafla',
    'severe bruising', 'michubuko mikali',
    'multiple people sick', 'watu wengi wagonjwa',
    'animal die off', 'wanyama wamekufa',
]

def is_severe_report(symptoms: str) -> bool:
    text = symptoms.lower()
    return any(signal in text for signal in SEVERE_SIGNALS)

@router.post("/log")
async def log_symptom(data: SymptomLog, db: Session = Depends(get_db)):
    """Lightweight endpoint — just saves symptom for outbreak tracking, no Claude call"""
    severe = is_severe_report(data.symptoms)
    entry = SymptomReport(
        region=data.region or "",
        symptoms=data.symptoms[:500],
        timestamp=datetime.fromisoformat(data.timestamp) if data.timestamp else datetime.utcnow(),
        flagged_severe=severe,
    )
    db.add(entry)
    db.commit()
    return {"success": True, "flagged_severe": severe}

@router.get("/ping")
async def ping():
    return {"status": "ok"}

@router.get("/severe")
async def get_severe_reports(admin: Admin = Depends(get_current_admin), db: Session = Depends(get_db)):
    """Returns recent symptom reports flagged as needing urgent human review."""
    reports = db.query(SymptomReport).filter(
        SymptomReport.flagged_severe == True
    ).order_by(SymptomReport.timestamp.desc()).limit(50).all()
    return {
        "reports": [
            {
                "id": r.id,
                "region": r.region,
                "symptoms": r.symptoms,
                "timestamp": r.timestamp.isoformat(),
            } for r in reports
        ],
        "total": len(reports),
    }

@router.delete("/{report_id}")
async def delete_symptom_report(report_id: int, admin: Admin = Depends(get_current_admin), db: Session = Depends(get_db)):
    """Admin-only deletion of a symptom report — e.g. spam, duplicate, or false flag.
    Permanently removes it from outbreak detection calculations."""
    report = db.query(SymptomReport).filter(SymptomReport.id == report_id).first()
    if not report:
        return {"success": False, "error": "Report not found"}
    db.delete(report)
    db.commit()
    return {"success": True}

@router.post("/{report_id}/dismiss")
async def dismiss_severe_flag(report_id: int, admin: Admin = Depends(get_current_admin), db: Session = Depends(get_db)):
    """Admin reviewed the severe flag and determined it's not urgent —
    clears the flag but keeps the report for outbreak statistics."""
    report = db.query(SymptomReport).filter(SymptomReport.id == report_id).first()
    if not report:
        return {"success": False, "error": "Report not found"}
    report.flagged_severe = False
    db.commit()
    return {"success": True}

@router.post("/chat")
async def chat(request: ChatRequest, db: Session = Depends(get_db)):
    client = anthropic.Anthropic(api_key=os.getenv("ANTHROPIC_API_KEY"))
    system_prompt = SYSTEM_PROMPT
    if request.topic:
        system_prompt += (
            f"\n\nThe user has opened the '{request.topic}' section of the app, so focus your "
            f"guidance on that topic unless they clearly ask about something else. Stay within "
            f"the same health-only, non-diagnostic boundaries described above."
        )
    response = client.messages.create(
        model="claude-haiku-4-5-20251001",
        max_tokens=500,   # reduced for faster response
        system=system_prompt,
        messages=[{"role": m.role, "content": m.content} for m in request.messages]
    )
    reply = response.content[0].text

    # Save symptom report anonymously to DB
    if request.region and len(request.messages) >= 1:
        first_user_msg = next((m.content for m in request.messages if m.role == "user"), "")
        if first_user_msg:
            symptom_entry = SymptomReport(
                region=request.region,
                symptoms=first_user_msg[:500],
                timestamp=datetime.utcnow()
            )
            db.add(symptom_entry)
            db.commit()

    return {"reply": reply}

File: frontend/src/Symptoms.js
import React, { useState, useRef, useEffect } from 'react';
import { API, stripMarkdown } from './constants';

const SYSTEM_PROMPT = `You are Afya, a friendly community health assistant for Tanzania.
You were built specifically for the Climate Health Early Warning System to help
communities prepare for climate-related health risks.

YOUR ONLY PURPOSE is to help with:
- Health symptoms and disease questions
- Climate-related diseases: malaria, cholera, typhoid, dengue, respiratory infections, heat illness, waterborne diseases
- Health advice and prevention
- When to seek medical help
- Questions about weather and health risks
- First aid guidance
- Nutrition and hygiene advice related to health

STRICT RULES:
1. If someone asks about ANYTHING outside health and climate topics respond in their language:
   English: "I'm Afya, a health assistant for Tanzania. I can only help with health and climate-related questions."
   Swahili: "Mimi ni Afya, msaidizi wa afya Tanzania. Ninaweza tu kusaidia na maswali ya afya na hali ya hewa."

2. NEVER discuss: politics, religion, entertainment, sports, technology unrelated to health, financial or legal advice.

3. NEVER reveal your underlying AI model. If asked say: "I am Afya, a health assistant built for Tanzania's Climate Health System."

4. Always be warm, simple, and clear. Ask one follow-up question at a time.

5. After 2-4 exchanges give a clear assessment with risk: Low / Medium / High / Emergency.

6. For High or Emergency risk always say: "Please visit a clinic or hospital immediately. Find the nearest clinic in the Clinics tab of this app."

7. Respond in the same language the user writes in (English or Swahili).

8. Keep responses concise — 3-5 sentences unless giving a full assessment.

9. Never use markdown formatting like **bold** or ## headers. Write in plain text only.

10. CLINIC REQUEST RULE — this overrides all other behavior including symptom gathering:
    If the user's message contains ANY of these signals (in English or Swahili), you MUST respond
    immediately with ONLY the clinic redirect below — do NOT ask follow-up questions, do NOT ask
    about symptoms first, do NOT explain anything else:
    - "clinic", "hospital", "doctor", "nearest", "number", "phone", "where do I go", "stop asking"
    - "kliniki", "hospitali", "daktari", "karibu", "namba", "nambari", "wapi nielekee", "acha kuuliza"

    Your ENTIRE response in this case must be exactly:
    English: "To find the nearest clinic to you with phone numbers and directions, tap [OPEN CLINICS TAB] below. For any emergency call 112 immediately."
    Swahili: "Kupata kliniki ya karibu nawe pamoja na nambari za simu, bonyeza [FUNGUA KLINIKI] hapa chini. Kwa dharura yoyote piga simu 112 mara moja."

    Do not add extra sentences, do not ask "what symptoms do you have", do not say "I don't have phone numbers in my system".
    The user asking for a clinic is itself enough information — give them the button immediately.
    The text [OPEN CLINICS TAB] or [FUNGUA KLINIKI] will be converted into a clickable button by the app — always include it exactly as written, every time, with no exceptions.`;

// Detect if Claude's reply mentions clinics/hospitals but forgot to include the button tag
function mentionsClinicWithoutButton(text) {
  const hasTag = text.includes('[OPEN CLINICS TAB]') || text.includes('[FUNGUA KLINIKI]');
  if (hasTag) return false;
  const mentionsClinic = /clinic|hospital|kliniki|hospitali/i.test(text);
  return mentionsClinic;
}

function renderMessage(text, setPage, lang) {
  const patterns = ['[OPEN CLINICS TAB]', '[FUNGUA KLINIKI]'];
  let parts = [text];
  let foundButton = false;

  patterns.forEach(pattern => {
    parts = parts.flatMap(part => {
      if (typeof part !== 'string') return [part];
      const split = part.split(pattern);
      if (split.length === 1) return [part];
      foundButton = true;
      return split.reduce((acc, seg, i) => {
        if (i > 0) acc.push(
          <button key={`${pattern}-${i}`}
            onClick={() => setPage && setPage('clinics')}
            style={{ display:'inline-flex', alignItems:'center', gap:4, background:'#2563eb', color:'#fff', border:'none', borderRadius:8, padding:'5px 12px', fontSize:13, fontWeight:600, cursor:'pointer', margin:'4px 0' }}>
            🏥 {pattern === '[FUNGUA KLINIKI]' ? 'Fungua Kliniki' : 'Open Clinics'}
          </button>
        );
        if (seg) acc.push(seg);
        return acc;
      }, []);
    });
  });

  // Safety net — if Claude mentioned clinic/hospital but forgot the button, add it anyway
  if (!foundButton && mentionsClinicWithoutButton(text)) {
    parts.push(
      <div key="clinic-fallback-btn" style={{ marginTop: 8 }}>
        <button onClick={() => setPage && setPage('clinics')}
          style={{ display:'inline-flex', alignItems:'center', gap:4, background:'#2563eb', color:'#fff', border:'none', borderRadius:8, padding:'5px 12px', fontSize:13, fontWeight:600, cursor:'pointer' }}>
          🏥 {lang === 'sw' ? 'Fungua Kliniki' : 'Open Clinics'}
        </button>
      </div>
    );
  }

  return <span>{parts}</span>;
}

// Single chat function — tries Vercel proxy first (fastest, no cold start)
// then falls back to Render backend
async function callAnthropicDirect(messages, timeoutMs = 20000) {
  // Call Anthropic API directly from the browser
  // REACT_APP_ prefix makes it available at build time in React
  const key = process.env.REACT_APP_ANTHROPIC_API_KEY || '';
  if (!key) { console.log('[Afya] direct: no REACT_APP_ANTHROPIC_API_KEY set, skipping'); return null; }
  const ctrl = new AbortController();
  const t = setTimeout(() => ctrl.abort(), timeoutMs);
  try {
    const res = await fetch('https://api.anthropic.com/v1/messages', {
      method: 'POST',
      signal: ctrl.signal,
      headers: {
        'Content-Type': 'application/json',
        'x-api-key': key,
        'anthropic-version': '2023-06-01',
        'anthropic-dangerous-direct-browser-access': 'true',
      },
      body: JSON.stringify({
        model: 'claude-haiku-4-5-20251001',
        max_tokens: 500,
        system: SYSTEM_PROMPT,
        messages: messages.map(m => ({ role: m.role, content: m.content })),
      }),
    });
    clearTimeout(t);
    if (!res.ok) { console.log('[Afya] direct: bad status', res.status, await res.text()); return null; }
    const data = await res.json();
    const text = data.content?.[0]?.text;
    if (!text) console.log('[Afya] direct: no text in response', data);
    return text ? stripMarkdown(text) : null;
  } catch (err) {
    clearTimeout(t);
    console.log('[Afya] direct: threw', err);
    return null;
  }
}

async function callVercelProxy(messages, timeoutMs = 15000) {
  const ctrl = new AbortController();
  const t = setTimeout(() => ctrl.abort(), timeoutMs);
  try {
    const res = await fetch(`${window.location.origin}/api/chat`, {
      method: 'POST',
      signal: ctrl.signal,
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ messages, system: SYSTEM_PROMPT }),
      cache: 'no-store',
    });
    clearTimeout(t);
    if (!res.ok) { console.log('[Afya] proxy: bad status', res.status, await res.text()); return null; }
    const data = await res.json();
    if (!data.reply) console.log('[Afya] proxy: no reply in response', data);
    return data.reply ? stripMarkdown(data.reply) : null;
  } catch (err) {
    clearTimeout(t);
    console.log('[Afya] proxy: threw', err);
    return null;
  }
}

async function callRenderBackend(messages, district, timeoutMs = 45000) {
  const ctrl = new AbortController();
  const t = setTimeout(() => ctrl.abort(), timeoutMs);
  try {
    const res = await fetch(`${API}/api/symptoms/chat`, {
      method: 'POST',
      signal: ctrl.signal,
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ messages, region: district }),
      cache: 'no-store',
    });
    clearTimeout(t);
    if (!res.ok) { console.log('[Afya] backend: bad status', res.status, await res.text()); return null; }
    const data = await res.json();
    if (!data.reply) console.log('[Afya] backend: no reply in response', data);
    return data.reply ? stripMarkdown(data.reply) : null;
  } catch (err) {
    clearTimeout(t);
    console.log('[Afya] backend: threw', err);
    return null;
  }
}

async function askAfya(messages, district, onStatus) {
  console.log('[Afya] askAfya called with messages:', messages);
  onStatus && onStatus('thinking');

  // Warm up Render in background
  fetch(`${API}/`).catch(() => {});

  // Route 1: Direct Anthropic API (fastest, works on desktop/modern mobile)
  const direct = await callAnthropicDirect(messages, 15000);
  if (direct) { console.log('[Afya] direct succeeded'); return direct; }

  // Route 2: Vercel serverless proxy (no CORS issues, same domain)
  onStatus && onStatus('retry-2');
  const proxy = await callVercelProxy(messages, 15000);
  if (proxy) { console.log('[Afya] proxy succeeded'); return proxy; }

  // Route 3: Render backend (final fallback)
  onStatus && onStatus('backend');
  const backend = await callRenderBackend(messages, district, 45000);
  if (backend) console.log('[Afya] backend succeeded');
  else console.log('[Afya] all 3 routes failed');
  return backend;
}

export default function Symptoms({ t, lang, district, setPage }) {
  const [messages, setMessages]         = useState([{ role:'assistant', content:t.afyaGreet }]);
  const [input, setInput]               = useState('');
  const [loading, setLoading]           = useState(false);
  const [emergency, setEmergency]       = useState(false);
  const [selectedTags, setSelectedTags] = useState([]);
  const [waitMsg, setWaitMsg]           = useState('');
  const bottomRef = useRef(null);
  const sw = lang === 'sw';

  useEffect(() => {
    bottomRef.current?.scrollIntoView({ behavior:'smooth' });
  }, [messages, loading]);

  // Wake up the Render backend as soon as this page opens, so it's warm
  // by the time the user finishes picking symptoms and hits send.
  useEffect(() => {
    fetch(`${API}/`).catch(() => {});
  }, []);

  function toggleTag(tag) {
    const updated = selectedTags.includes(tag)
      ? selectedTags.filter(t => t !== tag)
      : [...selectedTags, tag];
    setSelectedTags(updated);
    setInput(updated.join(', '));
  }

  async function saveHistory(text) {
    const hist = JSON.parse(localStorage.getItem('afya_symptom_history') || '[]');
    hist.push({ symptoms: text, district, date: new Date().toISOString() });
    localStorage.setItem('afya_symptom_history', JSON.stringify(hist.slice(-20)));
    try {
      const res = await fetch(`${API}/api/symptoms/log`, {
        method:'POST', headers:{'Content-Type':'application/json'},
        body: JSON.stringify({ symptoms:text, region:district, timestamp:new Date().toISOString() }),
      });
      const data = await res.json();
      return !!data.flagged_severe;
    } catch { return false; }
  }

  async function send(retryMessages = null) {
    const isRetry = !!retryMessages;
    let newMessages;
    console.log('[Afya] send() called, input:', JSON.stringify(input), 'isRetry:', isRetry);

    if (!isRetry) {
      if (!input.trim() || loading) { console.log('[Afya] send() bailed early - empty input or already loading'); return; }
      const userMsg = { role:'user', content:input };
      newMessages = [...messages, userMsg];
      setMessages(newMessages);
      const isSevere = await saveHistory(input);
      if (isSevere) setEmergency(true);
      setInput('');
      setSelectedTags([]);
    } else {
      newMessages = retryMessages;
    }

    setLoading(true);
    setWaitMsg(sw ? 'Afya anaangalia dalili zako...' : 'Afya is reviewing your symptoms...');

    const reply = await askAfya(newMessages, district, (status) => {
      if (status === 'thinking')        setWaitMsg(sw ? 'Afya anaangalia dalili zako...' : 'Afya is reviewing your symptoms...');
      else if (status.startsWith('retry')) setWaitMsg(sw ? 'Inajaribu tena...' : 'Retrying...');
      else if (status === 'backend')    setWaitMsg(sw ? 'Afya anafikiria...' : 'Afya is thinking...');
    });

    setWaitMsg('');
    setLoading(false);

    if (reply) {
      setMessages(prev => {
        const cleaned = prev.filter(m => !m._isError);
        return [...cleaned, { role:'assistant', content:reply }];
      });
      if (/emergency|hospital.*now|immediately|go.*now/i.test(reply)) setEmergency(true);
    } else {
      setMessages(prev => [
        ...prev.filter(m => !m._isError),
        {
          role:'assistant',
          content: sw
            ? 'Samahani, Afya haiwezi kujibu sasa hivi. Tafadhali angalia muunganisho wako au jaribu baadaye.'
            : 'Sorry, Afya is temporarily unavailable. Please check your connection or try again later.',
          _isError: true,
          _retryMessages: newMessages,
        }
      ]);
    }
  }

  return (
    <div style={{ padding:16, display:'flex', flexDirection:'column', height:'calc(100vh - 130px)' }}>
      <div style={{ fontSize:16, fontWeight:700, marginBottom:8 }}>🤒 {t.symptoms}</div>

      {emergency && (
        <div style={{ background:'#fef2f2', border:'1px solid #fecaca', borderRadius:10, padding:'9px 12px', marginBottom:8, display:'flex', justifyContent:'space-between', alignItems:'center' }}>
          <div style={{ fontSize:13, fontWeight:600, color:'#ef4444' }}>🚨 {sw?'Nenda hospitalini SASA':'Go to hospital NOW'}</div>
          <a href="tel:112" style={{ background:'#ef4444', color:'#fff', padding:'5px 12px', borderRadius:8, fontSize:12, textDecoration:'none', fontWeight:600 }}>112</a>
        </div>
      )}

      {/* Symptom tags */}
      <div style={{ display:'flex', gap:5, flexWrap:'wrap', marginBottom:6 }}>
        {t.symptomTags.map((tag, i) => {
          const sel = selectedTags.includes(tag);
          return (
            <button key={i} onClick={() => toggleTag(tag)}
              style={{ padding:'4px 9px', borderRadius:99, fontSize:11, cursor:'pointer', transition:'all 0.15s',
                background:sel?'#2563eb':'#f3f4f6', border:sel?'1px solid #2563eb':'1px solid #e5e7eb',
                color:sel?'#fff':'#374151', fontWeight:sel?600:400 }}>
              {sel ? '✓ ' : ''}{tag}
            </button>
          );
        })}
      </div>
      {selectedTags.length > 0 && (
        <div style={{ fontSize:11, color:'#6b7280', marginBottom:6, display:'flex', justifyContent:'space-between' }}>
          <span>✓ {selectedTags.length} {sw?'zimechaguliwa':'selected'}</span>
          <button onClick={() => { setSelectedTags([]); setInput(''); }}
            style={{ fontSize:11, color:'#ef4444', background:'none', border:'none', cursor:'pointer' }}>
            {sw?'Futa zote':'Clear all'}
          </button>
        </div>
      )}

      {/* Messages */}
      <div style={{ flex:1, overflowY:'auto', marginBottom:10 }}>
        {messages.map((m, i) => (
          <div key={i} style={{ display:'flex', justifyContent:m.role==='user'?'flex-end':'flex-start', marginBottom:8 }}>
            {m.role==='assistant' && (
              <div style={{ width:28, height:28, borderRadius:'50%', background:'#eff6ff', display:'flex', alignItems:'center', justifyContent:'center', fontSize:14, marginRight:6, flexShrink:0, alignSelf:'flex-end' }}>🤖</div>
            )}
            <div style={{ maxWidth:'80%' }}>
              <div style={{ padding:'9px 13px', borderRadius:16,
                background: m._isError ? '#fef2f2' : m.role==='user' ? '#2563eb' : '#fff',
                color: m._isError ? '#991b1b' : m.role==='user' ? '#fff' : '#111',
                border: m._isError ? '1px solid #fecaca' : m.role==='user' ? 'none' : '1px solid #e5e7eb',
                fontSize:14, lineHeight:1.55,
                borderBottomRightRadius:m.role==='user'?4:16,
                borderBottomLeftRadius:m.role==='assistant'?4:16 }}>
                {m.role==='assistant' ? renderMessage(m.content, setPage, lang) : m.content}
              </div>
              {/* Retry button for failed messages */}
              {m._isError && m._retryMessages && (
                <button
                  onClick={() => send(m._retryMessages)}
                  disabled={loading}
                  style={{ marginTop:4, fontSize:12, color:'#2563eb', background:'#eff6ff', border:'1px solid #bfdbfe', borderRadius:8, padding:'5px 12px', cursor:'pointer', fontWeight:600 }}>
                  🔄 {sw ? 'Jaribu Tena' : 'Try Again'}
                </button>
              )}
            </div>
          </div>
        ))}
        {loading && (
          <div style={{ display:'flex', alignItems:'center', marginBottom:8, gap:6 }}>
            <div style={{ width:28, height:28, borderRadius:'50%', background:'#eff6ff', display:'flex', alignItems:'center', justifyContent:'center', fontSize:14, flexShrink:0 }}>🤖</div>
            <div style={{ background:'#fff', border:'1px solid #e5e7eb', borderRadius:16, borderBottomLeftRadius:4, padding:'10px 14px' }}>
              {waitMsg ? (
                <div style={{ fontSize:12, color:'#6b7280' }}>{waitMsg}</div>
              ) : (
                <div style={{ display:'flex', gap:4, alignItems:'center' }}>
                  <span style={{ width:7, height:7, borderRadius:'50%', background:'#9ca3af', display:'inline-block', animation:'bounce 1.2s ease-in-out infinite' }} />
                  <span style={{ width:7, height:7, borderRadius:'50%', background:'#9ca3af', display:'inline-block', animation:'bounce 1.2s ease-in-out 0.2s infinite' }} />
                  <span style={{ width:7, height:7, borderRadius:'50%', background:'#9ca3af', display:'inline-block', animation:'bounce 1.2s ease-in-out 0.4s infinite' }} />
                </div>
              )}
            </div>
          </div>
        )}
        <div ref={bottomRef} />
      </div>

      <div style={{ display:'flex', gap:8 }}>
        <input value={input} onChange={e => setInput(e.target.value)}
          onKeyDown={e => e.key==='Enter' && !e.shiftKey && send()}
          placeholder={t.typeSymptoms}
          style={{ flex:1, padding:'10px 13px', borderRadius:10, border:`1px solid ${input?'#2563eb':'#e5e7eb'}`, fontSize:14, background:'#fff' }} />
        <button onClick={() => send()} disabled={loading||!input.trim()}
          style={{ padding:'10px 15px', background:input.trim()&&!loading?'#2563eb':'#e5e7eb', color:'#fff', border:'none', borderRadius:10, cursor:input.trim()&&!loading?'pointer':'default', fontSize:18 }}>➤</button>
      </div>

      <style>{`@keyframes bounce { 0%,80%,100%{transform:translateY(0)} 40%{transform:translateY(-6px)} }`}</style>
    </div>
  );
}
